conn.go

```diff
@@ -426,16 +426,14 @@ func encodeXtext(raw string) string {
 	out.Grow(len(raw))
 
 	for _, ch := range raw {
-		if ch == '+' || ch == '=' {
+		switch {
+		case ch > '!' && ch < '~' && ch != '+' && ch != '=':
```

Comment:
The RFC says:

```
      xchar = any ASCII CHAR between "!" (33) and "~" (126) inclusive,
              except for "+" and "=".
```

Shouldn't we use `ch >= '!' && ch <= '~'`  here?

Reply:
Wow. That's a terrible mistake. I'll fix and re-push it.

Reply:
It's not _that_ bad, it just means we will unnecessarily use xtext encoding for these characters as well. But yeah, it's better if we only use xtext encoding when necessary.

Reply:
However any less tolerant implementations might not accept sequences like `+21`.

Reply:
Fixed.